roles/openshift_certificate_expiry/tasks/main.yml

```diff
@@ -0,0 +1,25 @@
+---
+- name: Check cert expirys on host
+  openshift_cert_expiry:
+    warning_days: "{{ openshift_certificate_expiry_warning_days|int }}"
+    config_base: "{{ openshift_certificate_expiry_config_base }}"
+    show_all: "{{ openshift_certificate_expiry_show_all|bool }}"
+  register: check_results
+
+- name: Generate expiration report HTML
+  become: no
+  run_once: yes
+  template:
+    src: cert-expiry-table.html.j2
+    dest: "{{ openshift_certificate_expiry_html_report_path }}"
+  delegate_to: localhost
+  when: "{{ openshift_certificate_expiry_generate_html_report|bool }}"
+
+- name: Generate expiration results JSON
+  become: no
+  run_once: yes
+  template:
+    src: save_json_results.j2
```

Comment:
I think the maintenance costs are the same down the line. If the check results datastructures change then you either have to update a python file or a jinja template file.

I see what you're saying there though. Some of the template logic makes me feel a little gross and the serializing could look prettier w/ some actual python.

I will take a swing at making it into a role filter (I'm **keeping** this role self-contained). I'll also fix some typos I saw in the module code comments which are *killing me* on the inside.